### PR TITLE
routes: add missing support for Location header in /subscriptions

### PR DESCRIPTION
**Pull request by andrewbonney**
This is a simple fix for a missing header. The fact I end up using jsonify breaks the tests which pass in a mock object, so I still need to resolve that before this is merge-able.

### CODE REVIEW

File: nmosquery/common/routes.py [hunk 1]
```diff
@@ -85,7 +85,12 @@ def __subscriptions_post(self):
             elif data["secure"] is False:
                 abort(400, "All subscriptions must be secure when operating with HTTPS")
         obj, created = self.query.post_ws_subscribers(data)
-        return (201 if created else 200, obj)
+        response = make_response(jsonify(obj), 201 if created else 200)
+        response.autocorrect_location_header = False
+        response.headers["Location"] = "/x-nmos/query/{}/subscriptions/{}".format(
```
Comment (dannymeloy) [inline nmosquery/common/routes.py]: *potential knit-pick warning* - the Location Header only really serves a purpose for the 201 response code, not sure if it is misleading having it for a 200 response.. but probz doesn't matter.

File: nmosquery/common/routes.py [hunk 2]
```diff
@@ -85,7 +85,12 @@ def __subscriptions_post(self):
             elif data["secure"] is False:
                 abort(400, "All subscriptions must be secure when operating with HTTPS")
         obj, created = self.query.post_ws_subscribers(data)
-        return (201 if created else 200, obj)
+        response = make_response(jsonify(obj), 201 if created else 200)
+        response.autocorrect_location_header = False
```
Comment (dannymeloy) [inline nmosquery/common/routes.py]: Why are we choosing to use relative url's here? Apparently RFC 2616 says the location header should be an absolute path.

File: nmosquery/common/routes.py (hunk 1 above)
Comment (andrewbonney) [reply nmosquery/common/routes.py]: Agreed. This is just what the spec says to do, and does ensure some form of consistency for clients if they don't know a subscription already exists.

File: nmosquery/common/routes.py (hunk 2 above)
Comment (andrewbonney) [reply nmosquery/common/routes.py]: The API typically runs behind a reverse proxy, and I'm not sure we have the relevant code in place to adjust these absolute paths on the fly (possibly a consideration for any new webapi...). This matches the behaviour in the Registration API.

Comment (JamesGibo) [reply nmosquery/common/routes.py]: I believe RFC 2616 has been replaced by RFC7231, which allows both relative and absolute.
https://tools.ietf.org/html/rfc7231#section-7.1.2

Comment (dannymeloy) [reply nmosquery/common/routes.py]: Check you out Jimmy Gibbs with ya RFC-foo. PWNED.

File: tests/test_api.py [hunk 3]
```diff
@@ -205,29 +208,40 @@ def test_el_id(self, request, abort):
...
+            make_response.return_value.headers = {}
             self.queries[v].post_ws_subscribers.reset_mock()
-            self.queries[v].post_ws_subscribers.return_value = (mock.sentinel.obj, False)
-            self.assert_route_returns_value(v, '/x-nmos/query/' + v + '/subscriptions', [], (200, mock.sentinel.obj), request, method="POST")
+            self.queries[v].post_ws_subscribers.return_value = (test_obj, False)
+            self.assert_route_exists(path, request.method)
+            self.UUT.routes[path][request.method][0](*args)
             self.queries[v].post_ws_subscribers.assert_called_once_with(data)
 
             request.get_data = mock.MagicMock(return_value="{")
```
Comment (JamesGibo) [inline tests/test_api.py]: Should this have a closing bracket?

Comment (andrewbonney) [reply tests/test_api.py]: I don't think so, that's just setting up to ensure an exception is thrown when invalid data is used.

**Pull request merged**